modules/exploits/linux/smtp/harakiri.rb

```diff
@@ -0,0 +1,256 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+#
+# Exploit Title: Harakiri
+# ShortDescription: Haraka comes with a plugin for processing attachments. Versions before 2.8.9 can be vulnerable to command injection
+# Exploit Author: xychix [xychix at hotmail.com] / [mark at outflank.nl]
+# Date: 26 January 2017
+# Category: Remote Code Execution
+# Vendor Homepage: https://haraka.github.io/
+# Vendor Patch: https://github.com/haraka/Haraka/pull/1606
+# Software Link: https://github.com/haraka/Haraka
+# Exploit github: http://github.com/outflankbv/Exploits/
+# Vulnerable version link: https://github.com/haraka/Haraka/releases/tag/v2.8.8
+# Version:  <= Haraka 2.8.8 (with attachment plugin enabled)
+# Tested on: Should be OS independent tested on Ubuntu 16.04.1 LTS
+# Tested versions: 2.8.8 and 2.7.2
+# CVE : CVE-2016-1000282
+# Credits to: smfreegard for finding and reporting the vulnerability
+# Thanks to: Dexlab.nl for asking me to look at Haraka.
+#
+# Disclaimer:
+# This software has been created purely for the purposes of academic research and
+# for the development of effective defensive techniques, and is not intended to be
+# used to attack systems except where explicitly authorized. Project maintainers
+# are not responsible or liable for misuse of the software. Use responsibly.
+#
+# This is to be considered a responsible disclosure due to the availability of an effective patch.
+#
+#this code is based on:
+# https://raw.githubusercontent.com/rapid7/metasploit-framework/master/modules/exploits/linux/smtp/exim4_dovecot_exec.rb
+
+require 'msf/core'
+require 'base64'
+require 'zip'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Smtp
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Haraka Command Injection (used for Linux Payload download)',
+      'Description'    => %q{
+        This module exploits a command injection vulnerability in Haraka < 2.8.9 with
+        attachment plugin enabled and working.
+      },
+      'Author'         =>
+        [
+          'xychix / Mark, Outflank' # Exploit-DB python exploit and msf module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2016-1000282'],
+          [ 'URL', 'https://github.com/outflankbv/Exploits/blob/master/harakiri-CVE-2016-1000282.py'],
+          [ 'URL', 'https://www.exploit-db.com/exploits/41162/'],
+          [ 'URL', 'https://github.com/distributedweaknessfiling/DWF-Database-Artifacts/blob/158c10cf11bc7d6ad728c1a8dd213f523ecfca52/DWF/2016/1000282/CVE-2016-1000282.json'],
+          [ 'EDB-ID', '41162']
+        ],
+      'Privileged'     => false,
```

Comment:
Haraka can (and should!) be run as non root. However the default quick install won't do that.

https://haraka.github.io/manual/CoreConfig.html
writes:

> user - optionally a user to drop privileges to. Can be a string or UID.
> group - optionally a group to drop privileges to. Can be a string or GID.